pkg/resourceselector/blockdeviceselect/blockdeviceselector.go

```diff
@@ -0,0 +1,90 @@
+package blockdeviceselect
+
+import (
+	"fmt"
+	"github.com/openebs/node-disk-manager/cmd/ndm_daemonset/controller"
+	apis "github.com/openebs/node-disk-manager/pkg/apis/openebs/v1alpha1"
+	"github.com/openebs/node-disk-manager/pkg/resourceselector/verify"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BlockDeviceSelector selects a single block device from a list of block devices
+func (c *Config) BlockDeviceSelector(bdList *apis.BlockDeviceList) (*apis.BlockDevice, error) {
```

Comment:
The behaviour as understood from comments versus the method name does not match.

Can this be `FilterFrom`?
The caller code should look like
```go
import (
  blockdeviceselect "github.com/openebs/ndm/pkg/select/blockdevice"
)

config := blockdeviceselect.NewConfig()
bd, err := config.FilterFrom(bdList *apis.BlockDeviceList)
```

Reply:
The structure named `Config` does not really go along with `select` logic.

If I think from caller's perspective, I would like to have logic similar to below snippet _(this is more w.r.t readability)_:
```go
import (
  blockdeviceselect "github.com/openebs/ndm/pkg/select/blockdevice"
)

bdselect := blockdeviceselect.New()
bd, err := bdselect.From(bdList *apis.BlockDeviceList)
```

Alternative way to thin about this logic:
```go
import (
  blockdeviceselect "github.com/openebs/ndm/pkg/select/blockdevice"
)

bdselect := blockdeviceselect.From(bdList *apis.BlockDeviceList)
bd, err := bdselect.One()
```

Reply:
@AmitKumarDas  changed the name of the single entry method into the package. The 2nd and 3rd approach needs all method calls originating from the reconcilation loop. The package is designed such that, invoking a single method with BlockDeviceList, gives the suitable BlockDevice to be used.